plugins/woocommerce/src/Blocks/Domain/Services/CheckoutLink.php

```diff
@@ -94,21 +94,70 @@ protected function get_products_from_checkout_link() {
 		$raw_products = array_filter( explode( ',', wc_clean( wp_unslash( $_GET['products'] ?? '' ) ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 		$products     = [];
 
-		foreach ( $raw_products as $product_id_qty ) {
-			if ( strpos( $product_id_qty, ':' ) !== false ) {
-				list( $product_id, $qty ) = explode( ':', $product_id_qty );
+		foreach ( $raw_products as $raw_product ) {
+			$cart_item_data = [];
+
+			if ( strpos( $raw_product, ':' ) !== false ) {
+
+				list( $product_id, $qty, $product_data ) = array_pad( explode( ':', $raw_product, 3 ), 3, false );
```

Comment:
This is quite hard to read. Before it was fine because it was a simple `list` with 2 items. Maybe avoid list now and use explode only.

Reply:
Hold on, that is a regression since `&key=value` is no longer parsed into the `$exploded[2]`. I think I need the `{}` you mentioned [here](https://github.com/woocommerce/woocommerce/pull/59341#discussion_r2242106105)

Reply:
@mikejolley The `&` is breaking and I don't think we can nest it like this for the `&key=value` pairs. Should we go back to the `;` or do you have another preference?